src/com/amazon/ion/impl/_Private_IonReaderBuilder.java

```diff
@@ -0,0 +1,172 @@
+package com.amazon.ion.impl;
+
+import com.amazon.ion.IonException;
+import com.amazon.ion.IonReader;
+import com.amazon.ion.IonTextReader;
+import com.amazon.ion.IonValue;
+import com.amazon.ion.system.IonReaderBuilder;
+import com.amazon.ion.util.IonStreamUtils;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+
+import static com.amazon.ion.impl.LocalSymbolTable.DEFAULT_LST_FACTORY;
+import static com.amazon.ion.impl._Private_IonReaderFactory.makeIncrementalReader;
+import static com.amazon.ion.impl._Private_IonReaderFactory.makeReader;
+
+/**
+ * {@link IonReaderBuilder} extension for internal use only.
+ */
+public class _Private_IonReaderBuilder extends IonReaderBuilder {
+
+    private _Private_LocalSymbolTableFactory lstFactory;
+
+    private _Private_IonReaderBuilder() {
+        super();
+        lstFactory = DEFAULT_LST_FACTORY;
+    }
+
+    private _Private_IonReaderBuilder(_Private_IonReaderBuilder that) {
+        super(that);
+        this.lstFactory = that.lstFactory;
+    }
+
+    /**
+     * Declares the {@link _Private_LocalSymbolTableFactory} to use when constructing applicable readers.
+     *
+     * @param factory the factory to use, or {@link LocalSymbolTable#DEFAULT_LST_FACTORY} if null.
+     *
+     * @return this builder instance, if mutable;
+     * otherwise a mutable copy of this builder.
+     *
+     * @see #setLstFactory(_Private_LocalSymbolTableFactory)
+     */
+    public IonReaderBuilder withLstFactory(_Private_LocalSymbolTableFactory factory) {
+        _Private_IonReaderBuilder b = (_Private_IonReaderBuilder) mutable();
+        b.setLstFactory(factory);
+        return b;
+    }
+
+    /**
+     * @see #withLstFactory(_Private_LocalSymbolTableFactory)
+     */
+    public void setLstFactory(_Private_LocalSymbolTableFactory factory) {
+        mutationCheck();
+        if (factory == null) {
+            lstFactory = DEFAULT_LST_FACTORY;
+        } else {
+            lstFactory = factory;
+        }
+    }
+
+    public static class Mutable extends _Private_IonReaderBuilder {
+
+        public Mutable() {
+        }
+
+        public Mutable(IonReaderBuilder that) {
+            super((_Private_IonReaderBuilder) that);
+        }
+
+        @Override
+        public IonReaderBuilder immutable() {
+            return new _Private_IonReaderBuilder(this);
+        }
+
+        @Override
+        public IonReaderBuilder mutable() {
+            return this;
+        }
+
+        @Override
+        protected void mutationCheck() {
+        }
+
+    }
+
+    @Override
+    public IonReader build(byte[] ionData, int offset, int length)
```

Comment:
The `build` implementations are moved from IonReaderBuilder. I'll call out the one change below.